plugin/pkg/auth/authenticator/password/keystone/keystone.go

```diff
@@ -40,23 +44,49 @@ func (keystoneAuthenticator *KeystoneAuthenticator) AuthenticatePassword(usernam
 		Password:         password,
 	}
 
-	_, err := openstack.AuthenticatedClient(opts)
+	_, err := AuthenticatedClient(opts, keystoneAuthenticator.caFile)
 	if err != nil {
-		glog.Info("Failed: Starting openstack authenticate client")
+		glog.Info("Failed: Starting openstack authenticate client:" + err.Error())
 		return nil, false, errors.New("Failed to authenticate")
 	}
 
 	return &user.DefaultInfo{Name: username}, true, nil
 }
 
+// AuthenticatedClient logs in to an OpenStack cloud found at the identity endpoint specified by options, acquires a
+// token, and returns a Client instance that's ready to operate.
+func AuthenticatedClient(options gophercloud.AuthOptions, caFile string) (*gophercloud.ProviderClient, error) {
+	client, err := openstack.NewClient(options.IdentityEndpoint)
+	if err != nil {
+		return nil, err
+	}
+
+	config := &tls.Config{}
+	if caFile !="" {
+		roots, err := certutil.NewPool(caFile)
+		if err != nil {
+			return nil, err
+		}
+		config.RootCAs = roots
+	}
+	client.HTTPClient.Transport = &http.Transport{TLSClientConfig: config, }
```

Comment:
The client created by default `openstack.AuthenticatedClient` is `*gophercloud.ProviderClient`.

```
	return &gophercloud.ProviderClient{
		IdentityBase:     base,
		IdentityEndpoint: "",
	}, nil
```

The default `HTTPClient.Transport` is `nil`, and `DefaultTransport` is used instead.

```
var DefaultTransport RoundTripper = &Transport{
	Proxy: ProxyFromEnvironment,
	Dial: (&net.Dialer{
		Timeout:   30 * time.Second,
		KeepAlive: 30 * time.Second,
	}).Dial,
	TLSHandshakeTimeout:   10 * time.Second,
	ExpectContinueTimeout: 1 * time.Second,
}
``` 

The `client.HTTPClient.Transport` I used here is quite a new `Transport` with only `TLSClientConfig ` configured.

Do you suggest that we should merge these two together ?